storage/innobase/row/row0ins.cc

```diff
@@ -2658,6 +2657,34 @@ row_ins_clust_index_entry_low(
 				goto skip_bulk_insert;
 			}
 
+#ifdef WITH_WSREP
+			if (trx->is_wsrep() && wsrep_thd_is_local(trx->mysql_thd))
```

Comment:
The bulk of the added code needs to be moved to a new function, such as
```c++
#ifdef WITH_WSREP
ATTRIBUTE_COLD static bool row_ins_wsrep_start_bulk(trx_t *trx, const dict_index_t &index)
{ … }
#endif
```
The call would then be as follows:
```c++
if (trx->is_wsrep() && row_ins_wsrep_start_bulk(trx, *index)) {
	goto commit_exit;
}
```
This would add a minimum amount of overhead to the code for the common case that bulk insert is not being used.

Reply:
Sure, I will do this change on 10.6 first, it seems there is not many conflicts on 10.7